### PR TITLE
Update Rest API tests to no longer rely on FAB auth manager. Move tests specific to FAB permissions to FAB provider

### PR DESCRIPTION
**Pull request by vincbeck**
Many tests in `tests/api_connexion/endpoints/` rely on FAB permission model. Since we want to get rid of FAB, we need to make these tests agnostic from FAB. In this PR, they are now using simple auth manager.

I did two things:
- Any test that do not need granular permissions, keep them in `tests/api_connexion/endpoints/` and make them agnostic from FAB
- Any test using granular permissions, move them to FAB provider

This PR is quite big but I could not find a way to make it smaller. Here are some tips for a quicker review:
- Code change in `tests/api_connexion/endpoints/` is removing tests that use granular permissions
- Code change in `tests/providers/fab/auth_manager/api_endpoints/` is tests being copy pasted from `tests/api_connexion/endpoints/` that use granular permissions (the ones that are removed in the previous step)

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/models/dagrun.py [hunk 1]
```diff
@@ -1687,7 +1687,7 @@ class DagRunNote(Base):
 
     __tablename__ = "dag_run_note"
 
-    user_id = Column(Integer, nullable=True)
+    user_id = Column(String(100), nullable=True)
```
Comment (jscheffl) [inline airflow/models/dagrun.py]: Why do you make the user a lenght of 100 and the DB column in migration length of 128?

File: airflow/migrations/versions/0034_3_0_0_update_user_id_type.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+revision = "44eabb1904b4"
+down_revision = "16cbcb1c8c36"
+branch_labels = None
+depends_on = None
+airflow_version = "3.0.0"
+
+
+def upgrade():
+    with op.batch_alter_table("dag_run_note") as batch_op:
+        batch_op.alter_column("user_id", type_=sa.String(length=128))
```
Comment (jscheffl) [inline airflow/migrations/versions/0034_3_0_0_update_user_id_type.py:43]: As you migrate from INT to String, I _assume_ the new persistence is NOT the user ID from FAB anymore but the user name string. Correct? If yes, then I see no docs in the DB model classes and I would miss the conversion of UID to string - not that the previous INT is "just" stored as string?

**Review by jscheffl — COMMENTED** [review]
Two questions, once this is secured Okay for me to approve :-D

File: airflow/models/dagrun.py (hunk 1 above)
Comment (vincbeck) [reply airflow/models/dagrun.py]: Good catch! I'll fix that

File: airflow/migrations/versions/0034_3_0_0_update_user_id_type.py (hunk 2 above)
Comment (vincbeck) [reply airflow/migrations/versions/0034_3_0_0_update_user_id_type.py:43]: Nop. The user ID is provided by the auth manager. When this is FAB auth manager, the value does not change and remain the user ID (in this case, Integer is converted to String). But we cannot assume all auth managers will provide a user ID as integer, that's why I came up with String. This is a bug I found out when I was working on these tests.

File: airflow/models/dagrun.py (hunk 1 above)
Comment (vincbeck) [reply airflow/models/dagrun.py]: Fixed

**Review by jscheffl — APPROVED** [review]

**Comment by potiuk** [comment]
WHOA! Fantastic.

**Pull request merged**